portfolio/src/main/webapp/script.js

```diff
@@ -37,15 +37,38 @@ function addRandomQuote() {
  * This function opens the tab we want to access and hides the unnecessary tabs
  */
 function openTab(evt, tabName) {
-    var tabcontent = document.getElementsByClassName("tabcontent");
+    const tabcontent = document.getElementsByClassName("tabcontent");
     for(var i = 0; i <tabcontent.length; i++) {
```

Comment:
So kaleab was right here, better to use `let i = 0` than var.
Let vs. var is one of the super weird things about javascript, but essentially var is considered the old style, you basically always want to use let instead now.
Whats super weird about this is how they each handle scoping.
If you run
```
for (var i = 0; i < 3; i++) {}
console.log(i)
```
it will print `3`
if instead you run
```
for (let i = 0; i < 3; i++) {}
console.log(i);
```
youll get a variable not defined error.

You can read more about this and some of the other differences between let and var here: https://stackoverflow.com/questions/762011/whats-the-difference-between-using-let-and-var
But thats why this worked when you didn't redeclare i before.  Its also why it still works when you are redeclaring i, since var allows you to redeclare variables.  Let is more strict in how it can be used, but thats typically a good thing because it forces you to make sure you know what your variables are doing.

Reply:
Got it, thanks!